File: backend/entity/Comment.go
package entity

import (
	"gorm.io/gorm"
)

// ตาราง Comment
type Comment struct {
	gorm.Model
	Descript    string        `json:"descript"`
	Score       int           `json:"score"`
	UserComment []UserComment `gorm:"foreignKey:CommentID"`
}

type UserComment struct {
	gorm.Model
	User1ID   uint `json:"user1_id"`
	User2ID   uint `json:"user2_id"`
	CommentID uint `json:"comment_id"`
}

File: frontend/src/models/Comment.ts
import { UserInterface } from "./User";

export interface CommentInterface {
    ID: number;
    Descript: string;
    Score: number;
}

export interface UserCommentInterface {
    ID: number;
    User1ID: number;
    User1: UserInterface;
    User2ID: number;
    User2: UserInterface;
    CommentID: number;
    Comment: CommentInterface
}